internal/internal_decision_state_machine.go

```diff
@@ -114,6 +115,7 @@ type (
 	}
 
 	decisionsHelper struct {
+		counter          int64
```

Comment:
Can we rename this to something more meaningful? Isn't it "nextDecisionEventId"?